modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbTableStorage.java

```diff
@@ -171,21 +172,31 @@ private void checkPartitionId(int partitionId) {
     }
 
     /** {@inheritDoc} */
-    @Override public void destroyTxStateStorage(int partitionId) throws StorageException {
+    @Override public CompletableFuture<Void> destroyTxStateStorage(int partitionId) throws StorageException {
         checkPartitionId(partitionId);
 
-        TxStateStorage storage = storages.get(partitionId);
+        TxStateStorage storage = storages.getAndSet(partitionId, null);
+
+        if (storage == null) {
+            return completedFuture(null);
+        }
+
+        storage.destroy();
 
-        if (storage != null) {
+        return awaitFlush(false).whenComplete((v, e) -> {
             try {
                 storage.close();
-            } catch (Exception e) {
-                throw new StorageException("Couldn't close the transaction state storage of partition "
-                        + partitionId + ", table " + tableCfg.value().name());
+            }
+            catch (Exception ex) {
+                LOG.error("Couldn't close the transaction state storage of partition "
+                    + partitionId + ", table " + tableCfg.value().name());
             }
 
-            storage.destroy();
-        }
+            if (e != null) {
```

Comment:
I think that a thread that handles the future will print log message if it's needed.